test/infrastructure/test-extension/topologymutation/handler.go

```diff
@@ -0,0 +1,408 @@
+			if err := patchKubeadmControlPlaneTemplate(ctx, obj, variables); err != nil {
+				log.Error(err, "Error patching KubeadmControlPlaneTemplate")
+				return errors.Wrap(err, "error patching KubeadmControlPlaneTemplate")
+			}
+		case *bootstrapv1.KubeadmConfigTemplate:
+			if err := patchKubeadmConfigTemplate(ctx, obj, variables); err != nil {
+				log.Error(err, "Error patching KubeadmConfigTemplate")
+				return errors.Wrap(err, "error patching KubeadmConfigTemplate")
+			}
+		case *infrav1.VSphereClusterTemplate:
+			if err := patchGovmomiClusterTemplate(ctx, obj, variables); err != nil {
+				log.Error(err, "Error patching VSphereClusterTemplate")
+				return errors.Wrap(err, "error patching VSphereClusterTemplate")
+			}
+		case *infrav1.VSphereMachineTemplate:
+			if err := patchGovmomiMachineTemplate(ctx, obj, variables, isControlPlane); err != nil {
+				log.Error(err, "Error patching VSphereMachineTemplate")
+				return errors.Wrap(err, "error patching VSphereMachineTemplate")
+			}
+		case *vmwarev1.VSphereClusterTemplate:
+			if err := patchSupervisorClusterTemplate(ctx, obj, variables); err != nil {
+				log.Error(err, "Error patching VSphereClusterTemplate")
+				return errors.Wrap(err, "error patching VSphereClusterTemplate")
+			}
+		case *vmwarev1.VSphereMachineTemplate:
+			if err := patchSupervisorMachineTemplate(ctx, obj, variables, isControlPlane); err != nil {
+				log.Error(err, "Error patching VSphereMachineTemplate")
+				return errors.Wrap(err, "error patching VSphereMachineTemplate")
+			}
+		}
+		return nil
+	})
+}
+
+// patchKubeadmControlPlaneTemplate patches the KubeadmControlPlaneTemplate.
+func patchKubeadmControlPlaneTemplate(_ context.Context, tpl *controlplanev1.KubeadmControlPlaneTemplate, templateVariables map[string]apiextensionsv1.JSON) error {
+	// patch enableSSHIntoNodes
+	if err := patchUsers(&tpl.Spec.Template.Spec.KubeadmConfigSpec, templateVariables); err != nil {
+		return err
+	}
+
+	// patch kubeVipPodManifest
+	kubeVipPodManifest, err := topologymutation.GetStringVariable(templateVariables, "kubeVipPodManifest")
+	if err != nil {
+		// Skip patch if sshKey variable is not set
+		if topologymutation.IsNotFoundError(err) {
+			return nil
+		}
+		return err
+	}
+	controlPlaneIPAddr, err := topologymutation.GetStringVariable(templateVariables, "controlPlaneIpAddr")
+	if err != nil {
+		return err
+	}
+	kubeVipPodManifestModified := regexp.MustCompile("(name: address\n +value:).*").ReplaceAllString(kubeVipPodManifest, fmt.Sprintf("$1 %s", controlPlaneIPAddr))
+
+	for _, file := range kubevip.Files() {
+		if file.Path == "/etc/kubernetes/manifests/kube-vip.yaml" {
+			file.Content = kubeVipPodManifestModified
+		}
+		tpl.Spec.Template.Spec.KubeadmConfigSpec.Files = append(tpl.Spec.Template.Spec.KubeadmConfigSpec.Files, file)
+	}
+
+	// patch preKubeadmScript
+	preKubeadmScript, err := topologymutation.GetStringVariable(templateVariables, "preKubeadmScript")
```

Comment:
Our regular CC sesems to skip if preKubeadmScript is not set, do we need the same?